k8s/profiles.go

```diff
@@ -0,0 +1,249 @@
+package k8s
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	v1 "github.com/openfaas/faas-netes/pkg/apis/openfaas/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	typedCorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const ProfileAnnotationKey = "com.openfaas.profile"
+
+// ProfileClient defines the interface for CRUD operations on profiles
+// and applying faas-netes profiles to function Deployments.
+type ProfileClient interface {
+	Get(ctx context.Context, namespace string, names ...string) ([]Profile, error)
+}
+
+// Profile is and openfaas api extensions that can be predefined and applied
+// to functions by annotating them with `com.openfaas.profile: name1,name2`
+type Profile v1.ProfileSpec
+
+type profileCMClient struct {
+	kube typedCorev1.ConfigMapsGetter
+}
+
+// Get returns the named profiles, if found, from the namespace
+func (c profileCMClient) Get(ctx context.Context, namespace string, names ...string) ([]Profile, error) {
+	var resp []Profile
+	for _, name := range names {
+		cm, err := c.kube.ConfigMaps(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		profile := Profile{}
+
+		data := strings.NewReader(cm.Data["profile"])
+		err = yaml.NewYAMLOrJSONDecoder(data, 100).Decode(&profile)
```

Comment:
What does the magic number "100" stand for? Can you document and extract to a variable/const?

Reply:
The documentation of the method is pretty clear, should i copy it inline here?

> // NewYAMLOrJSONDecoder returns a decoder that will process YAML documents
> // or JSON documents from the given reader as a stream. bufferSize determines
> // how far into the stream the decoder will look to figure out whether this
> // is a JSON stream (has whitespace followed by an open brace).

Something like "use first 100 bytes to determine yaml or json parsing"